src/node/internal/crypto_keys.ts

```diff
@@ -380,48 +478,69 @@ export function createPrivateKey(
   key: CreateAsymmetricKeyOptions
 ): PrivateKeyObject;
 export function createPrivateKey(
-  _key: CreateAsymmetricKeyOptions | KeyData
+  key: CreateAsymmetricKeyOptions | KeyData
 ): PrivateKeyObject {
-  // The options here are fairly complex. The key data can be a string,
-  // ArrayBuffer, or ArrayBufferView. The first argument can be one of
-  // these or an object with a key property that is one of these. If the
-  // key data is a string, then it will be decoded using an encoding
-  // (defaults to UTF8).
-  throw new ERR_METHOD_NOT_IMPLEMENTED('crypto.createPrivateKey');
-  // return KeyObject.from(cryptoImpl.createPrivateKey(
-  //     validateAsymmetricKeyOptions(key, kPrivateKey))) as PrivateKeyObject;
+  const cryptoKey = cryptoImpl.createPrivateKey(
+    prepareAsymmetricKey(key, KeyContext.kCreatePrivate)
+  );
+  return KeyObject.from(cryptoKey) as PrivateKeyObject;
 }
 
 export function createPublicKey(key: string): PublicKeyObject;
 export function createPublicKey(key: ArrayBuffer): PublicKeyObject;
 export function createPublicKey(key: ArrayBufferView): PublicKeyObject;
-
 export function createPublicKey(key: KeyObject): PublicKeyObject;
 export function createPublicKey(key: CryptoKey): PublicKeyObject;
 export function createPublicKey(
   key: CreateAsymmetricKeyOptions
 ): PublicKeyObject;
 export function createPublicKey(
-  _key: CreateAsymmetricKeyOptions | KeyData | CryptoKey | KeyObject
+  key: CreateAsymmetricKeyOptions | KeyData | CryptoKey | KeyObject
 ): PublicKeyObject {
-  // The options here are a bit complicated. The key material itself can
-  // either be a string, ArrayBuffer, or ArrayBufferView. It is also
-  // possible to pass a private key in the form of either a CryptoKey
-  // or KeyObject. The first argument can be one of these, or an object
-  // whose key value is one of these. If the key data is a string, then
-  // it will be decoded using an encoding (defaults to UTF8). If a
-  // CryptoKey or KeyObject is passed, it will be used to derived the
-  // public key.
-  throw new ERR_METHOD_NOT_IMPLEMENTED('crypto.createPublicKey');
-  // return KeyObject.from(cryptoImpl.createPublicKey(
-  //     validateAsymmetricKeyOptions(key, kPublicKey))) as PublicKeyObject;
+  // Passing a KeyObject or a CryptoKey allows deriving the public key
+  // from an existing private key.
+
+  if (isKeyObject(key)) {
+    if (key.type !== 'private') {
+      throw new ERR_INVALID_ARG_TYPE('key', 'PrivateKeyObject', key);
+    }
+    return KeyObject.from(
+      cryptoImpl.createPublicKey({
+        key: (key as KeyObject)[kHandle],
```

Comment:
If line 503 isKeyObject call assertion works, we won't need this key as KeyObject check. I think there is a bug here...